File: BMI/app/src/main/java/com/example/tipper/MainActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable; // for EditText event handling
import android.text.TextWatcher; // EditText listener
import android.view.View;
import android.widget.Button;
import android.widget.EditText; // for bill amount input
import android.widget.TextView; // for displaying text


public class MainActivity extends AppCompatActivity {

    private double weight = 0.0;
    private double height = 0.15;
    //TextViews definitions
    private TextView heightTextView;
    private TextView weightTextView;


    private TextView totalTextView;

    //method onCreate, most important one
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState); // call superclass onCreate
        setContentView(R.layout.activity_main); // inflate the GUI

//looking up TextViews in layout
        weightTextView = (TextView) findViewById(R.id.weightTextView);
        heightTextView = (TextView) findViewById(R.id.heightTextView);
        totalTextView = (TextView) findViewById(R.id.totalTextView);
        totalTextView.setText(String.format(0 + ""));

        Button button_to_bh_calc = (Button) findViewById(R.id.button_to_bh_calc);
        Button button_to_recommendations = (Button) findViewById(R.id.button_to_recommendations);
        Button button_to_diagram = (Button) findViewById(R.id.button_to_diagram);
        Button button_to_quiz = (Button) findViewById(R.id.button_to_quiz);



        button_to_quiz.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                openQuiz();
            }
        });
        button_to_diagram.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                openDiagram();
            }
        });

        button_to_bh_calc.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                openCaloriesCalc();
            }
        });

        button_to_recommendations.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                openRecommendation();
            }
        });




// setting TextWatchers for both weight and height fields
        EditText weightEditText =
                (EditText) findViewById(R.id.weightEditText);
        weightEditText.addTextChangedListener(weightEditTextWatcher);
        EditText heightEditText =
                (EditText) findViewById(R.id.heightEditText);
        heightEditText.addTextChangedListener(heightEditTextWatcher);

    }

    //method calculate, calculating BMI, setting totalTextView as result
    private void calculate() {


        double total = weight / (height * height);


        totalTextView.setText(String.format(total + ""));
    }


    // listeners object for TextViews weight and height text-changed events
    private final TextWatcher weightEditTextWatcher = new TextWatcher() {
        // called when the user modifies weight or height field, or both
        @Override
        public void onTextChanged(CharSequence s, int start,
                                  int before, int count) {

            try {
                weight = Double.parseDouble(s.toString()) / 100.0;
                weightTextView.setText(String.format(weight + ""));
            } catch (NumberFormatException e) {
                weightTextView.setText("");
                weight = 0.0;
            }
//recalculate on change
            calculate();
        }

        @Override
        public void afterTextChanged(Editable s) {
        }

        @Override
        public void beforeTextChanged(
                CharSequence s, int start, int count, int after) {
        }
    };
    private final TextWatcher heightEditTextWatcher = new TextWatcher() {

        @Override
        public void onTextChanged(CharSequence s, int start,
                                  int before, int count) {

            try {
                height = Double.parseDouble(s.toString()) / 100.0;
                heightTextView.setText(String.format(height + ""));
            } catch (NumberFormatException e) {
                heightTextView.setText("");
                height = 0.0;
            }
//recalculate on change
            calculate();
        }

        @Override
        public void afterTextChanged(Editable s) {
        }

        @Override
        public void beforeTextChanged(
                CharSequence s, int start, int count, int after) {
        }
    };


    public void openCaloriesCalc(){
        Intent intent = new Intent(this, CaloriesCalc.class);
        startActivity(intent);
    }

    public void openQuiz(){
        Intent intent = new Intent(this, Quiz.class);
        startActivity(intent);
    }

    public void openRecommendation(){
        Intent intent = new Intent(this, Recommendations.class);
        startActivity(intent);
    }

    public void openDiagram(){
        Intent intent = new Intent(this, Diagram.class);
        startActivity(intent);
    }
}

File: BMI/app/src/main/java/com/example/tipper/MainActivity.kt
package com.example.tipper

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

// for EditText event handling
// EditText listener
// for bill amount input
// for displaying text
class MainActivity : AppCompatActivity() {
    private var weight = 0.0
    private var height = 0.15

    //TextViews definitions
    private var heightTextView: TextView? = null
    private var weightTextView: TextView? = null
    private var totalTextView: TextView? = null

    //method onCreate, most important one
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState) // call superclass onCreate
        setContentView(R.layout.activity_main) // inflate the GUI

//looking up TextViews in layout
        weightTextView = findViewById<View>(R.id.weightTextView) as TextView
        heightTextView = findViewById<View>(R.id.heightTextView) as TextView
        totalTextView = findViewById<View>(R.id.totalTextView) as TextView
        totalTextView!!.text = String.format(0.toString() + "")
        val button_to_bh_calc = findViewById<View>(R.id.button_to_bh_calc) as Button
        val button_to_recommendations = findViewById<View>(R.id.button_to_recommendations) as Button
        val button_to_diagram = findViewById<View>(R.id.button_to_diagram) as Button
        val button_to_quiz = findViewById<View>(R.id.button_to_quiz) as Button
        button_to_quiz.setOnClickListener { openQuiz() }
        button_to_diagram.setOnClickListener { openDiagram() }
        button_to_bh_calc.setOnClickListener { openCaloriesCalc() }
        button_to_recommendations.setOnClickListener { openRecommendation() }


// setting TextWatchers for both weight and height fields
        val weightEditText = findViewById<View>(R.id.weightEditText) as EditText
        weightEditText.addTextChangedListener(weightEditTextWatcher)
        val heightEditText = findViewById<View>(R.id.heightEditText) as EditText
        heightEditText.addTextChangedListener(heightEditTextWatcher)
    }

    //method calculate, calculating BMI, setting totalTextView as result
    private fun calculate() {
        val total = weight / (height * height)
        totalTextView!!.text = String.format(total.toString() + "")
    }

    // listeners object for TextViews weight and height text-changed events
    private val weightEditTextWatcher: TextWatcher = object : TextWatcher {
        // called when the user modifies weight or height field, or both
        override fun onTextChanged(s: CharSequence, start: Int,
                                   before: Int, count: Int) {
            try {
                weight = s.toString().toDouble() / 100.0
                weightTextView!!.text = String.format(weight.toString() + "")
            } catch (e: NumberFormatException) {
                weightTextView!!.text = ""
                weight = 0.0
            }
            //recalculate on change
            calculate()
        }

        override fun afterTextChanged(s: Editable) {}
        override fun beforeTextChanged(
                s: CharSequence, start: Int, count: Int, after: Int) {
        }
    }
    private val heightEditTextWatcher: TextWatcher = object : TextWatcher {
        override fun onTextChanged(s: CharSequence, start: Int,
                                   before: Int, count: Int) {
            try {
                height = s.toString().toDouble() / 100.0
                heightTextView!!.text = String.format(height.toString() + "")
            } catch (e: NumberFormatException) {
                heightTextView!!.text = ""
                height = 0.0
            }
            //recalculate on change
            calculate()
        }

        override fun afterTextChanged(s: Editable) {}
        override fun beforeTextChanged(
                s: CharSequence, start: Int, count: Int, after: Int) {
        }
    }

    fun openCaloriesCalc() {
        val intent = Intent(this, CaloriesCalc::class.java)
        startActivity(intent)
    }

    fun openQuiz() {
        val intent = Intent(this, Quiz::class.java)
        startActivity(intent)
    }

    fun openRecommendation() {
        val intent = Intent(this, Recommendations::class.java)
        startActivity(intent)
    }

    fun openDiagram() {
        val intent = Intent(this, Diagram::class.java)
        startActivity(intent)
    }
}